### PR TITLE
Properly handle move feedback if mouse up happens outside window

### PR DESCRIPTION
**Pull request by martin-fleck-at**
#### What it does

- Handle mouse down if we should be dragging as mouse up
- Reset feedback if selection has changed
-- Ensure element does not have to be selected to revert move
-- Add method for listeners to selection service

Fixes https://github.com/eclipse-glsp/glsp/issues/1426

#### How to test

See https://github.com/eclipse-glsp/glsp/issues/1426

#### Changelog

-   [ ] This PR should be mentioned in the changelog
-   [ ] This PR introduces a breaking change (if yes, provide more details below for the changelog and the migration guide)

### CODE REVIEW

**Review by ivy-lli — COMMENTED** [review]
Looks good.
Quickly tried the workflow example with this change -> seems good
Tried also our glsp project (via replace the @eclipse-glsp/client src and lib in the node_modules), but got an error (not sure if this is because of this change or something else?
image

File: packages/client/src/features/tools/change-bounds/change-bounds-tool-move-feedback.ts [hunk 1]
```diff
@@ -185,14 +196,14 @@ export class FeedbackMoveMouseListener extends DragAwareMouseListener {
         if (!this.tracker.isTracking()) {
             return [];
         }
-        const elementsToMove = this.getElementsToMove(target);
+        const elementToMove = this.getElementsToMove(target);
```
[the remark this answers is not in the document]
Comment (martin-fleck-at) [reply packages/client/src/features/tools/change-bounds/change-bounds-tool-move-feedback.ts:199]: Yeah, based on the wrong out-dated commit, good catch!

**Comment by martin-fleck-at** [comment]
> Looks good. Quickly tried the workflow example with this change -> seems good Tried also our glsp project (via replace the @eclipse-glsp/client src and lib in the node_modules), but got an error (not sure if this is because of this change or something else? image

Very interesting, do you already see that on the latest nightly? Because I don't think I changed anything regarding injection. As for the error itself, I wonder if the circular dependency is because of the `ISelectionService` in the file so moving that part to a separate file might help. I'll push an update to this PR soon which maybe you can try out. Otherwise we need to debug a little deeper.

**Comment by ivy-lli** [comment]
Seems to work with [2.3.0-next.389](https://www.npmjs.com/package/@eclipse-glsp/client/v/2.3.0-next.389)
However, I don't want to pretend that I did everything correctly with the lib/src replacement in the node_modules... Maybe I missed something here.

**Comment by tortmayr** [comment]
@ivy-lli I think this is indeed related to the `lib/src` replacement. I did a quick test and published the PR to a local npm registry (verdaccio). Consuming it from there seems to work without issues.

**Comment by tortmayr** [comment]
@martin-fleck-at There also seems to be an issue if I drag outside of the window during a move and then release the mouse.
Then the element gets not reseted and remains on the invalid position.
But I'm not sure if we can fix that easily.

**Comment by martin-fleck-at** [comment]
Based on the comment by Tobias, I just fixed the variable naming. I'd suggest we merge the PR if everyone is happy with it and then see if the consumption works and if not I can work on a fix since it is otherwise quite hard to test. What do you think?

**Review by tortmayr — APPROVED** [review]

**Pull request merged**